vassal-app/src/main/java/VASSAL/tools/ProblemDialog.java

```diff
@@ -353,27 +338,55 @@ public void run() {
 
     if (thrown != null) logger.error("", thrown);
 
-    return DialogUtils.enqueue(new Runnable() {
-      @Override
-      public void run() {
-        DetailsDialog.showDialog(
-          parent,
-          title,
-          heading,
-          message,
-          details,
-          Resources.getString("Dialogs.disable"),
-          Resources.getString("Dialogs.show_details"),
-          Resources.getString("Dialogs.hide_details"),
-          messageType,
-          key
-        );
-      }
-    });
+    return DialogUtils.enqueue(() -> DetailsDialog.showDialog(
+      parent,
+      title,
+      heading,
+      message,
+      details,
+      Resources.getString("Dialogs.disable"),
+      Resources.getString("Dialogs.show_details"),
+      Resources.getString("Dialogs.hide_details"),
+      messageType,
+      key
+    ));
   }
 
   private static Frame getFrame() {
     return GameModule.getGameModule() == null
       ? null : GameModule.getGameModule().getPlayerWindow();
   }
+
+  public static Future<?> showDeprecated(String date) {
+    final StackWalker.StackFrame frame =
+      StackWalker.getInstance(Set.of(StackWalker.Option.RETAIN_CLASS_REFERENCE), 2)
+        .walk(f -> f.skip(1).limit(1).collect(Collectors.toList())).get(0);
+    final String method = frame.getClassName() + "." + frame.getMethodName();
+    String expiry;
+    try {
+      expiry = LocalDate.parse(date, DateTimeFormatter.ofPattern("uuuu-M-d")).plusYears(1).format(DateTimeFormatter.ofPattern("d-MMM-uuuu", Resources.getLocale()));
+    }
+    catch (Exception ignored) {
```

Comment:
What's the actual exception type which could come out of the try block?

Reply:
Ah, I think that was left over from an earlier version. No longer required, I have removed the try/catch block.